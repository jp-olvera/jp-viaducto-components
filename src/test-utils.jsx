import React from 'react';
import { render } from '@testing-library/react';
import { ConfigProvider } from './providers';

const AllTheProviders = ({ children }) => {
  return <ConfigProvider>{children}</ConfigProvider>;
};

const customRender = (ui, options) =>
  render(ui, { wrapper: AllTheProviders, ...options });

// re-export everything
export * from '@testing-library/react';

// override render method
export { customRender as render };
