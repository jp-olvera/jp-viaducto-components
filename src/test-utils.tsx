import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import { ConfigProvider } from './providers';

const AllTheProviders = ({ children }: any) => (
  <ConfigProvider>{children}</ConfigProvider>
);

jest.mock('./cells/Input/amex.svg', () => null);
jest.mock('./cells/Input/mastercard.svg', () => null);
jest.mock('./cells/Input/visa.svg', () => null);
jest.mock('./cells/Dropdown/sorting.svg', () => null);
jest.mock('./cells/BareButton/close.svg', () => null);

const customRender = (ui, options?) => render(ui, { wrapper: AllTheProviders, ...options });
// re-export everything
export * from '@testing-library/react';

// override render method
export { customRender as render };
