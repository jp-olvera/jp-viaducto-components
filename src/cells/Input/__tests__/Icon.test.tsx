/* eslint-env jest */

import React from 'react';
import '@testing-library/jest-dom/extend-expect';
import { render } from '../../../test-utils';
import { Icon } from '..';

describe('<Icon/>', () => {
  test('should render null icon', () => {
    const { container } = render(<Icon icon='aaaaaaa' />);
    expect(container).toMatchSnapshot();
  });
  describe('render cards svgs', () => {
    test('should render visa svg', () => {
      const { container } = render(<Icon icon='visa' />);
      expect(container).not.toBeNull();
    });
    test('should render mastercard svg', () => {
      const { container } = render(<Icon icon='mastercard' />);
      expect(container).not.toBeNull();
    });
    test('should render american-express svg', () => {
      const { container } = render(<Icon icon='american-express' />);
      expect(container).not.toBeNull();
    });
  });
});
