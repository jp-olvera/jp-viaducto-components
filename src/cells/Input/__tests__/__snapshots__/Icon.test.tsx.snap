// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Icon/> should render null icon 1`] = `<div />`;
